crates/toml_edit/src/parser/datetime.rs

```diff
@@ -255,6 +255,7 @@ pub(crate) fn unsigned_digits<'i, const MIN: usize, const MAX: usize>(
     input: &mut Input<'i>,
 ) -> PResult<&'i str> {
     take_while(MIN..=MAX, DIGIT)
+        // Safety: `digit` only produces ASCII
         .map(|b: &[u8]| unsafe { from_utf8_unchecked(b, "`is_ascii_digit` filters out on-ASCII") })
```

Comment:
This is duplicating the debug assertion message.  Is that a specific special comment syntax you are needing?

The comment is stale from our conversion from `is_ascii_digit` to `DIGIT`

Reply:
@epage Yeah so this comment is twofold, for one I wasn't sure what the debug message was referring to, and the other thing is that it _is_ a bit of a norm to justify safety with a `Safety:` comment. I think the debug message mostly satisfies the need, but I got super confused as an unsafe reviewer so decided to just do it the proper way.

Reply:
I can understand the standard convention but we are double checking the invariants in debug builds and I want those panics to carry over what assumption is being broken, so we have the message already and I don't want to be doubling up on the message.  Or put another way, this is an extra level of protection over a simple comment.